doc/howtos/rdtraining/firstui.rst

```diff
@@ -0,0 +1,275 @@
+  (without having to know its in-database identifier).
+- ``model`` has a fixed value of ``ir.actions.act_window`` (:ref:`reference/actions/window`)
+- ``name`` is the name of the action.
+- ``res_model`` is the model to which the action applies.
+- ``view_mode`` are the views which will be available, in this case the list (tree) and form views.
+  We'll see `later <howto/rdtraining/qwebintro>` that there can be other view modes.
+
+Examples can be found everywhere in Odoo, but
+`this one <https://github.com/odoo/odoo/blob/09c59012bf80d2ccbafe21c39e604d6cfda72924/addons/crm/views/crm_lost_reason_views.xml#L57-L70>`__
+is a good example of simple action. It also gives you the structure of an XML data file.
+
+.. exercise:: Add an action.
+
+    Create the ``estate_property_views.xml`` file in the appropriate folder and define it in the
+    ``__manifest__.py`` file.
+
+    Create an action for the model ``estate.property``.
+
+Restart the server, you should see the file loaded in the log.
+
+Menus
+=====
+
+**Reference**: the documentation related to this topic can be found in
+:ref:`reference/data/shortcuts`.
+
+.. note::
+
+    **Goal**: at the end of this section, three menus should be created and the default view is
+    displayed:
+
+    .. image:: firstui/media/estate_menu_root.png
+      :align: center
+      :alt: Root menus
+
+    .. image:: firstui/media/estate_menu_action.png
+      :align: center
+      :alt: First level and action menus
+
+    .. image:: firstui/media/estate_form_default.png
+      :align: center
+      :alt: Default form view
+
+Because menus are somewhat complex to declare there is a ``<menuitem>``
+shortcut to declare an ``ir.ui.menu`` and connect it to the corresponding
+action more easily.
+
+A minimal menu for our ``test_model_action`` would look like:
+
+.. code-block:: xml
+
+    <menuitem id="test_model_menu_action" action="test_model_action"/>
+
+The menu ``test_model_menu_action`` is linked to the action ``test_model_action``, and the action
+is linked to the model ``test.model``. As mentioned before, the action can be seen as the link
+between the menu and the model.
+
+However, menus always follow an architecture; in practice, there are three levels of menus:
+
+1. The root menu, which is displayed in the App switcher
+2. The first level menu, displayed in the top bar
+3. The action menus
+
+   .. image:: firstui/media/menu_01.png
+      :align: center
+      :alt: Root menus
+
+   .. image:: firstui/media/menu_02.png
+      :align: center
+      :alt: First level and action menus
+
+The easiest way to define the structure is to create a structure in the XML file. A minimal
+structure for our ``test_model_action`` would look like:
+
+.. code-block:: xml
+
+    <menuitem id="test_menu_root" name="Test">
+        <menuitem id="test_first_level_menu" name="First Level">
+            <menuitem id="test_model_menu_action" action="test_model_action"/>
+        </menuitem>
+    </menuitem>
```

Comment:
I've tried to reproduce this structure and I got an error.
AssertionError: Element odoo has extra content: menuitem, line 2

```xml
<menuitem id="estate_menu_root" name="Real Estates">
    <menuitem id="estate_advertisments_menu" name="Advertisments">
        <menuitem id="estate_properties_menu_action" action="estate_properties_action"/>
    </menuitem>
</menuitem>
```

This worked:
```xml
<menuitem id="estate_menu_root" name="Real Estates"/>
<menuitem id="estate_advertisments_menu" name="Advertisments" parent="estate_menu_root"/>
<menuitem id="estate_properties_menu_action" action="estate_properties_action" parent="estate_advertisments_menu"/>
```

Reply:
Weird, this is the complete file:
```
<?xml version="1.0" encoding="utf-8"?>
<odoo>
    <!-- Top menu item -->
    <menuitem id="estate_menu_root" name="Real Estate">
        <menuitem id="estate_advertisement_menu" name="Advertisements" sequence="5">
            <menuitem id="estate_property_menu_action" action="estate_property_action" sequence="5"/>
        </menuitem>
        <menuitem id="estate_settings_menu" name="Settings" sequence="10">
            <menuitem id="estate_property_type_menu_action" action="estate_property_type_action" sequence="5"/>
            <menuitem id="estate_property_tag_menu_action" action="estate_property_tag_action" sequence="10"/>
        </menuitem>
    </menuitem>
</odoo>

```

Reply:
I didn't have any issue either:
`<odoo>
	<menuitem id="estate_property_root" name="Real Estate">
	    <menuitem id="estate_property_advertisements_menu" name="Advertisements">
	        <menuitem id="estate_property_menu_action" 
	        	name="Properties" action="estate_property_action"/>
	    </menuitem>
	    <menuitem id='estate_property_settings_menu' name="Settings">
	    	<menuitem id="estate_property_type_menu_action" action="estate_property_type_action"/>
	    	<menuitem id="estate_property_tag_menu_action" action="estate_property_tag_action"/>
	    </menuitem>
	</menuitem>
</odoo>`

Reply:
It appears to be working on `master`, but not on `13.0`.